unlock-js/src/v12/createLock.js

```diff
@@ -42,14 +42,20 @@ export default async function(lock, callback) {
   let currencyContractAddress = lock.currencyContractAddress || ZERO
 
   const lockName = lock.name || 'New Lock'
+
+  // Building a salt from the lock name will prevent creators from creating 2 locks with the same name.
```

Comment:
Just curious what the rationale is behind this

Reply:
The salt is used to compute the lock address, along with the user's nonce.
This will prevent someone from creating 2 locks at the same time with the same name. (like pressing "create lock" twice...)